src/sources/syslog.rs

```diff
@@ -219,59 +224,110 @@ impl SyslogDecoder {
         // |
         // | ASCII decimal number of unknown length
 
-        if let Some(i) = src.iter().position(|&b| b == b' ') {
-            let len: usize = std::str::from_utf8(&src[..i])
-                .map_err(|_| ())
-                .and_then(|num| num.parse().map_err(|_| ()))
-                .map_err(|_| {
-                    LinesCodecError::Io(io::Error::new(
-                        io::ErrorKind::InvalidData,
-                        "Unable to decode message len as number",
-                    ))
-                })?;
-
-            let from = i + 1;
-            let to = from + len;
-
-            if let Some(msg) = src.get(from..to) {
-                let s = std::str::from_utf8(msg)
-                    .map_err(|_| {
-                        LinesCodecError::Io(io::Error::new(
-                            io::ErrorKind::InvalidData,
-                            "Unable to decode message as UTF8",
-                        ))
-                    })?
-                    .to_string();
-                src.advance(to);
-                Ok(Some(s))
-            } else {
-                Ok(None)
+        loop {
+            let read_to = cmp::min(self.other.max_length().saturating_add(1), src.len());
```

Comment:
What is the purpose of the `+ 1` here?

Reply:
That's a good question. I'm not 100% sure. It comes from LinesCodec:

https://github.com/tokio-rs/tokio/blob/master/tokio-util/src/codec/lines_codec.rs#L114

Reply:
If I understand correctly: `max_length` refers to the maximum line length, excluding the `\n` character. To include `\n`, `1` is added.